pkg/compare/compare.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compare
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+
+	"sigs.k8s.io/yaml"
+
+	"istio.io/operator/pkg/tpath"
+)
+
+// YAMLCmpReporter is an option to generate tree based diff for YAMLs, used by cmp.Equal().
+type YAMLCmpReporter struct {
+	path     cmp.Path
+	diffTree map[string]interface{}
+}
+
+// PushStep implements Reporter.PushStep(PathStep) interface to keep track of current path by pushing
```

Comment:
It's idiomatic to just write, "<foo> implements <bar> interface." It's enough - the reader can look at the interface to find out more.

Reply:
Fixed.